outputs.tf

```
@@ -1,13 +1,14 @@
 locals {
   binding = try(google_pubsub_subscription_iam_binding.binding[0], null)
   member  = try(google_pubsub_subscription_iam_member.member, null)
+  policy  = try(google_pubsub_subscription_iam_policy.policy[0], null)
 
-  iam_output = [local.binding, local.member]
+  iam_output = [local.binding, local.member, local.policy]
 
-  iam_output_index = var.authoritative ? 0 : 1
+  iam_output_index = var.policy_bindings != null ? 2 : var.authoritative ? 0 : 1
```

```suggestion
  iam_output_index = var.policy_bindings != null && var.authoritative ? 2 : var.authoritative ? 0 : 1
```

Ignore if when policies are defined authoritative is ignored, although as commented before I think this could be confusing, but I have less context on this.

---

invalid change request ;) due to reasons ;) see other comment